_posts/2017-02-21-Rails-5-1-beta1.markdown

```diff
@@ -0,0 +1,108 @@
+
+We have a lovely, simple API for declaring new resource routes. But if you'd like to add new programmatic routes that has logic determining the final destination based on the parameters, well, you'd have to row your own boat with helpers and other messy approaches.
+
+With directed routes, you can now declare programmatic routes that has the full power of Ruby to do different things depending on the parameters passed.
+
+With resolved routes, you can reprogram the polymorphic look-up for models based straight to compatible methods. So this allow you to turn `link_to @comment` into a final route like `message_path(@comment.parent, anchor: "comment_#{@comment.id}")`.
+
+Thank you to [Andrew White](https://github.com/pixeltrix) for [making all this work](https://github.com/rails/rails/pull/23138)!
+
+
+## Unify form_tag/form_for with form_with
+
+We've long had two parallel structures for creating forms. Those that were based off records through form_for, were we used convention over configuration to extract the details, and manually configured ones using form_tag. Now we've unified these two hierarchies with [form_with](https://github.com/rails/rails/issues/25197). A single root tree that you can configure through a inferred record or manually. It's much nicer and simpler.
+
+Thanks to [Kasper Timm Hansen][kasper] for this one too!
+
+
+## Everything else
+
+In addition to the highlight reel, we have hundreds of other fixes and improvements across all the frameworks. Please peruse the CHANGELOGs to acquaint yourself with all the goodies:
+
+* [Action Cable CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/actioncable/CHANGELOG.md)
+* [Action Mailer CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/actionmailer/CHANGELOG.md)
+* [Action Pack CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/actionpack/CHANGELOG.md)
+* [Action View CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/actionview/CHANGELOG.md)
+* [Active Model CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/activemodel/CHANGELOG.md)
+* [Active Record CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/activerecord/CHANGELOG.md)
+* [Active Support CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/activesupport/CHANGELOG.md)
+* [Active Job CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/activejob/CHANGELOG.md)
+* [Railties CHANGELOG](https://github.com/rails/rails/blob/v5.1.0.beta1/railties/CHANGELOG.md)
+
+Your release manager for Rails 5.1 is [Rafael França][rafaelfranca]. He'll be chaperoning us through the betas, release candidates, and onto the final release in advance of [RailsConf 2017](http://railsconf.com/).
+
+As per our [maintenance policy][maintenance-policy], the release of Rails 5.1 will mean that bug fixes will only apply to 5.1.x, regular security issues to 5.1.x and 5.0.x, and severe security issues to 5.1.x, 5.0.x, and 4.2.x. This means 4.x and below will essentially be unsupported!
```

Comment:
> This means 4.x and below will essentially be unsupported!

Small note: is `4.x` here correct? 
The previous sentence says that severe security issues will still be released for `4.2.x`.
I'm not sure if that makes `4.2` still count as "essentially unsupported".